draft-ietf-lamps-csr-attestation.md

```diff
@@ -686,6 +645,78 @@ ext-ar EXTENSION ::= {
+point of view of the calling application. This section is particularly
+applicable to the background check model.
+
+Cases where the CSR is generated internally to the Attesting Environment
+are straightforward: the Hardware Security Model (HSM) generates and embeds
+the Evidence and the corresponding certification paths when constructing the CSR.
+
+Cases where the CSR is generated externally may require extra communication
+between the CSR generator and the Attesting Environment, first to obtain
+the necessary Evidence about the subject key, and then to use
+the subject key to sign the CSR; for example, a CSR generated by
+a popular crypto library about a subject key stored on a PKCS#11 {{PKCS11}} device.
+
+As an example, assuming that the HSM is, or contains, the Attesting Environment and
+some cryptographic library is assembling a CSR by interacting with the HSM over some
+network protocol, then the interaction would conceptually be:
+
+~~~ aasvg
+                   +---------+          +-----+
+                   | Crypto  |          | HSM |
+                   | Library |          |     |
+                   +---------+          +-----+
+                        |                  |
+                        | getEvidence()    |
+                        |----------------->|
+                        |                  |
+                        |<-----------------|
++---------------------+ |                  |
+| CSR = assembleCSR() |-|                  |
++---------------------+ |                  |
+                        |                  |
+                        | sign(CSR)        |
+                        |----------------->|
+                        |                  |
+                        |<-----------------|
+                        |                  |
+~~~
+{: #fig-csr-client-p11 title="Example interaction between CSR generator and HSM."}
+
+## Separation of RA and CA roles with respect to Attestation Results {#sec-impl-ar}
+
+As described in {{architecture}}, CSRs MAY contain either Evidence or Attestation Results (AR),
+and also the Registration Authority (RA) and Certification Authority (CA) MAY be conceptualized as
+a single Relying Party entity, or as separate entities. There are some implications here worth discussion.
+
+In many cases, the Evidence contained within a CSR is intended to be consumed by the RA and not
+to be placed into the issued certificate.
+In some RA / CA architectures, it MAY be appropriate for the RA to "consume" the Evidence
+and remove it from the CSR, re-signing the CSR with an RA signing key.
+The mechanics of doing this in a CRMF CSR are more complex since an RA is not allowed to modify
+the `certReq` field when a proof-of-possession `popo` field is present, so an additional CRMF control
+could be required to indicate that the `ext-evidence` or `ext-ar` extensions are not to be copied
+into the issued certificate. Such CRMF controls are not defined in this document.
```

Comment:
I think this sentence is wrong. CRMF can use for the ProofOfPossession choice raVerified. Instead PKCS#10 requires a self-signature as proof-of-posession.

Reply:
I simplified the paragraph based on your input